### PR TITLE
Transition the project to a Poetry-based setup and revitalize basic test automation

### PR DESCRIPTION
**Pull request by Nicoretti**
* Migrate project configuration to poetry
* Fix/setup basic CI infrastructure
* Drop support for outdated versions (DB + Python)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+name: 🐞 Bug Report
+about: File a bug report
+title: 🐞 <Insert Title>
+labels: bug
+assignees: ''
+
+---
+
+### Checklist
+In order to improve the time, how long a bugfix does take, please make sure you address as much as possible of the following checklist.
```
Comment (kaklakariada) [inline .github/ISSUE_TEMPLATE/bug.md:11]:
```suggestion
In order to speedup fixing the bug, please make sure you address as many items from the following checklist as possible.
```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+- [ ] I have reproduced the issue with at least the latest released version of \<PROJECT-NAME\>.
```
Comment (kaklakariada) [inline .github/ISSUE_TEMPLATE/bug.md:13]:
```suggestion
- [ ] I have reproduced the issue with the latest released version of pyexasol.
```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Describe the actual observed behaviour: A clear and concise description of what you observed.
+Screenshot: If applicable, add screenshots to help explaining your problem.
+-->
+
+### System
+<!---
+Describe the system and the used configuration.
+-->
+**Desktop:**
+ - OS: [e.g. Linux (Fedora 35), Linux (Ubuntu 20.01), MacOs (...), Windows (...)]
```
Comment (kaklakariada) [inline .github/ISSUE_TEMPLATE/bug.md:65]:
```suggestion
 - OS: [e.g. Linux (Fedora 35), Linux (Ubuntu 20.01), macOS (...), Windows (...)]
```

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
+
+    steps:
+    - name: SCM Checkout
+      uses: actions/checkout@v4
+
+    - name: Setup Python & Poetry Environment
+      uses: exasol/python-toolbox/.github/actions/python-environment@0.7.0
+      with:
+        python-version: ${{ matrix.python-version }}
```
Comment (kaklakariada) [inline .github/workflows/checks.yml:45]: It's a waste of energy to checkout the code twice and install dependencies twice. Better run integration tests after the unit tests.

File: .github/workflows/ci-master.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+name: Continues Integration (Master)
```
Comment (kaklakariada) [inline .github/workflows/ci-master.yml:1]:
```suggestion
name: Continuous Integration (Master)
```

File: .github/workflows/ci-pr.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+name: Continues Integration (PR)
```
Comment (kaklakariada) [inline .github/workflows/ci-pr.yml:1]:
```suggestion
name: Continuous Integration (PR)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## [Unreleased]
+
+- Switch packaging and project workflows to poetry
+- Drop support for python 3.7
+- Drop support for Exasol 6.x
+- Drop support for Exasol 7.0.x
```
Comment (kaklakariada) [inline CHANGELOG.md:8]: Is it necessary to add ticket links?

File: docs/REFERENCE.md [hunk 8]
```diff
@@ -567,7 +567,7 @@ Return `True` if view exists, `False` otherwise. If schema name was not specifie
 
 ### list_schemas()
 
-Return list of schemas from [EXA_SCHEMAS](https://docs.exasol.com/sql_references/metadata/metadata_system_tables.htm#EXA_SCHEMAS) system view matching LIKE-pattern.
+Return list of schemas from [EXA_SCHEMAS](https://docs.exasol.com/db/latest/sql_references/system_tables/metadata/exa_schemas.htm) system view matching LIKE-pattern.
```
Comment (kaklakariada) [inline docs/REFERENCE.md:570]: Think about adding a broken links checker, e.g. https://github.com/exasol/project-keeper/blob/main/.github/workflows/broken_links_checker.yml

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+python = "^3.8"
+websocket-client = ">=1.0.1"
+pyopenssl = "*"
+rsa = "*"
+packaging = "*"
+pandas = { version = "*", optional = true }
+ujson = { version = "*", optional = true }
+python-rapidjson = { version = "*", optional = true }
+orjson = { version = ">=3.6", optional = true }
+pproxy = { version = "*", optional = true }
```
Comment (kaklakariada) [inline pyproject.toml:52]: Are there tests that verify, that pyexasol also works without these dependencies?

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: .github/workflows/checks.yml (hunk 4 above)
Comment (Nicoretti) [reply .github/workflows/checks.yml:45]: It is faster this way, and the cost is the same or did I miss something else?

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**